go-sms_inbound.js

```diff
@@ -236,29 +240,42 @@ go.utils = {
             });
     },
 
-    // Create a new contact with the minimum required details
-    create_contact: function(msisdn, im) {
-        var payload = {
-            "details": {
+    // Create a new identity
+    create_identity: function(im, msisdn, communicate_through_id, operator_id) {
+        var payload = {};
+
+        // compile base payload
+        if (msisdn) {
+            payload.details = {
                 "default_addr_type": "msisdn",
                 "addresses": go.utils.get_addresses(msisdn)
-            }
-        };
+            };
+        }
+
+        if (communicate_through_id) {
+            payload.communicate_through = communicate_through_id;
+        }
+
+        // add operator_id if available
+        if (operator_id) {
+            payload.operator = operator_id;
+        }
+
         return go.utils
-            .service_api_call("identities", "post", null, payload, 'identities', im)
+            .service_api_call("identities", "post", null, payload, 'identities/', im)
             .then(function(json_post_response) {
                 var contact_created = json_post_response.data;
-                // Return the contact's id
-                return contact_created.id;
+                // Return the contact
+                return contact_created;
             });
     },
 
     // Gets a contact if it exists, otherwise creates a new one
-    get_or_create_contact: function(msisdn, im) {
+    get_or_create_identity: function(msisdn, im, operator_id) {
```

Comment:
lets make country code a config value and then passed into normalize function, else the value as a util drops